### DISCUSSION TITLE
Does Remix have dependency tracing like Next?

### DISCUSSION THREAD

**Discussion by WesleyYue** [remix-run/remix#7582] [Q&A] [answered]
Specifically, [this feature](https://nextjs.org/docs/pages/api-reference/next-config-js/output) to reduce deployment sizes by automatically tracing only the dependency packages required for the production build.

**Comment by sergiodxa** [comment, marked as answer]
Remix compiler will only bundle the dependencies your code import. If you want to remove from node_modules the development dependencies you can do `npm install --production`.
